samples/datatypes.js

```diff
@@ -0,0 +1,696 @@
+  // Imports the Google Cloud client library.
+  const {Spanner} = require('@google-cloud/spanner');
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'my-project-id';
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+
+  // Creates a client
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  // Gets a reference to a Cloud Spanner instance and database.
+  const instance = spanner.instance(instanceId);
+  const database = instance.database(databaseId);
+
+  const fieldType = {
+    type: 'string',
+  };
+
+  const exampleString = 'Venue 42';
+
+  const query = {
+    sql: `SELECT VenueId, VenueName FROM Venues
+            WHERE VenueName = @venueName`,
+    params: {
+      venueName: exampleString,
+    },
+    types: {
+      venueName: fieldType,
+    },
+  };
+
+  // Queries rows from the Venues table.
+  try {
+    const [rows] = await database.run(query);
+
+    rows.forEach(row => {
+      const json = row.toJSON();
+      console.log(`VenueId: ${json.VenueId}, VenueName: ${json.VenueName}`);
+    });
+  } catch (err) {
+    console.error('ERROR:', err);
+  } finally {
+    // Close the database when finished.
+    database.close();
+  }
+  // [END spanner_query_with_string_parameter]
+}
+
+async function queryWithTimestamp(instanceId, databaseId, projectId) {
+  // [START spanner_query_with_timestamp_parameter]
+  // Imports the Google Cloud client library.
+  const {Spanner} = require('@google-cloud/spanner');
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'my-project-id';
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+
+  // Creates a client.
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  // Gets a reference to a Cloud Spanner instance and database.
+  const instance = spanner.instance(instanceId);
+  const database = instance.database(databaseId);
+
+  const fieldType = {
+    type: 'timestamp',
+  };
+
+  const exampleTimestamp = new Date().toISOString();
+
+  const query = {
+    sql: `SELECT VenueId, VenueName, LastUpdateTime FROM Venues
+            WHERE LastUpdateTime < @lastUpdateTime`,
+    params: {
+      lastUpdateTime: exampleTimestamp,
+    },
+    types: {
+      lastUpdateTime: fieldType,
+    },
+  };
+
+  // Queries rows from the Venues table.
+  try {
+    const [rows] = await database.run(query);
+
+    rows.forEach(row => {
+      const json = row.toJSON();
+      console.log(
+        `VenueId: ${json.VenueId}, VenueName: ${json.VenueName},` +
+          ` LastUpdateTime: ${json.LastUpdateTime}`
+      );
+    });
+  } catch (err) {
+    console.error('ERROR:', err);
+  } finally {
+    // Close the database when finished.
+    database.close();
+  }
+  // [END spanner_query_with_timestamp_parameter]
+}
+
+require(`yargs`)
```

Comment:
we haven't been using yargs for our new sample format (the idea being that it can cause some confusion for customers, as they need to install a dependency).

Here's the approach we've been taking:

https://github.com/googleapis/nodejs-web-risk/blob/master/samples/quickstart.js#L46

Doesn't need to be a blocker from me, but I'd rather move towards consistency.